spec/controllers/post_controller_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+require 'rails-controller-testing'
+Rails::Controller::Testing.install
+
+RSpec.describe PostsController, type: :request do
```

Comment:
- [ ] Kindly make sure your both controller testing files are inside `spec/requests/` directory and not `spec/controllers/` directory as required for the project i.e. you should rename your `spec/controllers/` to `spec/requests/`